vm/src/pyobject.rs

```diff
@@ -948,44 +938,35 @@ pub trait DictProtocol {
     fn del_item(&self, key: &str);
 }
 
-impl DictProtocol for PyObjectRef {
-    fn contains_key(&self, k: &str) -> bool {
-        if let Some(dict) = self.payload::<PyDict>() {
-            objdict::content_contains_key_str(&dict.entries.borrow(), k)
-        } else {
-            unimplemented!()
-        }
-    }
-
-    fn get_item(&self, k: &str) -> Option<PyObjectRef> {
-        if let Some(dict) = self.payload::<PyDict>() {
-            objdict::content_get_key_str(&dict.entries.borrow(), k)
-        } else {
-            panic!("TODO {:?}", k)
-        }
-    }
+pub trait ItemProtocol {
```

Comment:
I think this should be the final shape of the `ItemProtocol`. It took 22 iterations, but I think this is the proper signature :).